packages/squiggle-lang/src/js/math.ts

```diff
@@ -0,0 +1,113 @@
+type SampleArgs = {
+  probs: number[];
+  size: number;
+};
+
+/** Searches for a first element in `arr` greater than `value` */
+function binsearchFirstGreater(arr: number[], value: number): number {
+  let s = 0
+  let e = arr.length - 1;
+  while ((e - s) > 1) {
+    const i = Math.ceil((s + e) / 2)
+    if (arr[i] < value) {
+      s = i + 1
+    } else {
+      e = i
+    }
+  }
+  return value < arr[s] ? s : e
+}
+
+export function randomSample(dist: number[], args: SampleArgs): number[] {
```

Comment:
just fyi, if it happens to be easy, it would be pretty useful to start allowing seeds for the low-level random functions

Reply:
Yes! I would try to add this as soon as I get to squiggle code from down here)